### DISCUSSION TITLE
Getting the highest-priority camera

### DISCUSSION THREAD

**Discussion by eisegesis-623** [ramokz/phantom-camera#344] [Question / Help Wanted] [answered]
Considering some relatively recent changes (including the introduction of the PhantomCameraManager), what is the best way to go about getting the highest-priority camera currently in the scene tree?

**Comment by ramokz** [comment, marked as answer]
It depends a bit on the scene structure and whether if you have both a `PCamHost` and the `PCams` in the scene, but you could for example do:
```gdscript
PhantomCameraManager.phantom_camera_hosts[0].get_active_pcam()
```
Which fetches the currently active `PCam`, i.e. the `PCam` with the highest priority in the scene.

---

Otherwise, if you only got `PCams` in a scene, you would currently have to filter the list by doing something like:
```gdscript
var pcams_by_priority: Array
pcams_by_priority = PhantomCameraManager.phantom_camera_3ds
pcams_by_priority.sort_custom(
  func (a: PhantomCamera3D, b: PhantomCamera3D):
    return a.priority > b.priority
)
```

Then you can just grab the first index item of the array with `pcams_by_priority[0]`.

Could probably make an internal public getter function that does the above to make that easier.

**Comment by eisegesis-623** [comment]
Thanks so much!
